src/main/java/org/openstreetmap/josm/plugins/pt_assistant/actions/PTMendRelationAction.java

```diff
@@ -0,0 +1,1483 @@
+            // OK.actionPerformed(null);
+            save();
+            if (currentIndex < members.size() - 1) {
+                notice = null;
+                callNextWay(currentIndex);
+            } else {
+                notice = null;
+                deleteExtraWays();
+            }
+        } else if (chr == 'B' || chr == '2') {
+            if (currentIndex < members.size() - 1) {
+                notice = null;
+                currentIndex = wayIndices.get(wayIndices.size() - 1);
+                callNextWay(currentIndex);
+            } else {
+                notice = null;
+                deleteExtraWays();
+            }
+        } else if (chr == 'C' || chr == '4') {
+            List<Command> cmdlst = new ArrayList<>();
+            int[] lst = wayIndices.stream().mapToInt(Integer::intValue).toArray();
+            for (int i = 0; i < lst.length; i++) {
+                Way w = members.get(lst[i]).getWay();
+                TagMap newKeys = w.getKeys();
+                newKeys.put("oneway", "bus=no");
+                cmdlst.add(new ChangePropertyCommand(Collections.singleton(w), newKeys));
+            }
+            UndoRedoHandler.getInstance().add(new SequenceCommand("Add tags", cmdlst));
+            // OK.actionPerformed(null);
+            save();
+            if (currentIndex < members.size() - 1) {
+                notice = null;
+                callNextWay(currentIndex);
+            } else {
+                notice = null;
+                deleteExtraWays();
+            }
+        }
+        if (chr == 'R' || chr == '3') {
+            // calculate the previous index
+            int prevIndex = -1;
+            for (int i = currentIndex - 1; i >= 0; i--) {
+                if (members.get(i).isWay()) {
+                    prevIndex = i;
+                    break;
+                }
+            }
+            // remove all the ways
+            int[] lst = wayIndices.stream().mapToInt(Integer::intValue).toArray();
+            memberTableModel.remove(lst);
+            for (int i = 0; i < lst.length; i++) {
+                members.remove(lst[i] - i);
+            }
+            // OK.actionPerformed(null);
+            save();
+            if (prevIndex != -1) {
+                notice = null;
+                callNextWay(prevIndex);
+            } else {
+                notice = null;
+                deleteExtraWays();
+            }
+        }
+    }
+
+
+    private void findWayafterchunkRoundabout(Way way) {
+        List<Node> breakNode = new ArrayList<>();
+        breakNode.add(currentNode);
+        splitNode = way.lastNode();
+        Strategy strategy = new TempStrategyRoundabout();
+        List<List<Node>> wayChunks = SplitWayCommand.buildSplitChunks(way, breakNode);
+        SplitWayCommand result = SplitWayCommand.splitWay(way, wayChunks, Collections.emptyList(), strategy);
+        if (result != null) {
+            UndoRedoHandler.getInstance().add(result);
+        }
+    }
```

Comment:
```suggestion
    private void findWayafterchunkRoundabout(final Way way) {
        final Strategy strategy = wayChunks ->
            StreamSupport.stream(wayChunks.spliterator(), false)
                .filter(w -> w.containsNode(way.lastNode()))
                .findFirst()
                .orElse(null);
        final SplitWayCommand result = SplitWayCommand.splitWay(
            way,
            SplitWayCommand.buildSplitChunks(way, Collections.singletonList(currentNode)),
            Collections.emptyList(),
            strategy
        );
        if (result != null) {
            UndoRedoHandler.getInstance().add(result);
        }
    }
```

This would get rid of the inner class `TempStrategyRoundabout` in `AbstractMendRelationAction` and also get rid of the field `protected Node splitNode;`, which is only used to pass the last node of the way to `TempStrategyRoundabout`.
Probably `TempStrategy` can be removed in a similar way.